scripts/ai_api_helper.py

```
@@ -63,21 +123,32 @@
+    # Use exponential backoff for reliability
+    def api_call():
+        if provider == 'anthropic':
+            return call_anthropic_api(prompt, max_tokens)
+        elif provider == 'gemini':
+            return call_gemini_api(prompt, max_tokens)
+        else:
+            return None, f"Unknown provider: {provider}"
+    
+    try:
+        return exponential_backoff_retry(api_call)
+    except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)